packages/svelte/src/internal/client/runtime.js

```diff
@@ -1226,18 +1226,21 @@ export function freeze(value) {
 		!(STATE_FROZEN_SYMBOL in value)
 	) {
 		// If the object is already proxified, then snapshot the value
-		if (STATE_SYMBOL in value) {
-			value = snapshot(value);
-		}
-		define_property(value, STATE_FROZEN_SYMBOL, {
+		var copy = STATE_SYMBOL in value ? snapshot(value) : value;
```

Comment:
Wondering if this is an unintended consequence of using `$state.frozen` on something that's proxified. I think I wouldn't expect to have the inner class objects etc be "destroyed".

In general, since `$state.frozen` in prod no longer freezes the object (for perf reasons), we should maybe stop doing that in dev (since people could do checks in dev mode that rely on that which then fail in prod), and in general not snapshot the value anymore, i.e. leave the state mostly alone.

Reply:
Yeah this isn't right. We need to preserve the classes here. In fact, we need the old behaviour back for this case and likely need some test coverage too.